internal/clients/http/azure/create_vm.go

```diff
@@ -0,0 +1,395 @@
+// strongly inspired by https://github.com/Azure-Samples/azure-sdk-for-go-samples/tree/main/sdk/resourcemanager/compute/create_vm
+// from commit 2ea51d9744d5d4680b4983b2e292a7b1d0085ff4
+
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/RHEnVision/provisioning-backend/internal/ptr"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+)
+
+var subscriptionId string
+
+const TraceName = "github.com/RHEnVision/provisioning-backend/internal/clients/http/azure"
+
+const (
+	resourceGroupName = "redhat-deployed2"
+	vmName            = "redhat-vm"
+	vnetName          = "redhat-vnet"
+	subnetName        = "redhat-subnet"
+	nsgName           = "redhat-nsg"
+	nicName           = "redhat-nic"
+	diskName          = "redhat-disk"
+	publicIPName      = "redhat-public-ip"
+	location          = "eastus"
+	adminUsername     = "azureuser"
+)
+
+func (c *client) CreateVM(ctx context.Context, imageID string, sshKeyBody string) (*armcompute.VirtualMachine, error) {
```

Comment:
Now I know this is very early draft, but could you start using our models early? I mean:

* Pubkey
* AzureInstanceType

Reply:
sure :+1: will do ;)